### PR TITLE
Closes #3 Adding contribution workflow commands to make it easier to minify CSS and SVG files.

### LINKED ISSUE #3: Create a more full-featured contribution workflow.
## Motivation
This is for use in the future CDN.

### PR DESCRIPTION
**Pull request by trackleft**
## Description
We need a way to minify files before they are deployed to the CDN.

## Related issues
Closes #3 

## How to test
Proof-read the CONTRIBUTING.md file

clone and checkout this branch of this repository
In the root directory of the repository, use **`npm i`** to install
List of commands to use in the **root** directory, and what they do...

Delete the dist folder
```
npm run clean
```
Make new dist folders to be populated later
```
npm run mkdirs
```
Copy css file to the dist folder
```
npm run copy:css
```
Copy fonts to the appropriate dist folder
```
npm run copy:fonts
```
Copy pngs to the appropriate dist folder
```
npm run copy:pngs
```
Minify and copy SVGs to the appropriate dist folder
```
npm run svg-minify
```
Minify the CSS within the **dist** folder and add a new file with .min.css suffix
```
npm run css-minify
```
Run all the copy commands in parallel (instead of each one individually)
```
npm run copy
```
Run all of the commands in the proper order
```
npm run dist
```
To test that you can install the correct files in a custom project from this PR
```
 npm install "https://github.com/az-digital/az-icons.git#issues/3-contribution-workflow" --save
```
Then check the node_modules/az-icons folder for the project files. It should look like this
README.md
dist/
package.json 

The dist folder should contain all of the files that exist in the dist folder of the project.

Additionally, I've updated the github actions to use the dist/. folder contents instead of the root contents.

Another step you can take to test that icons still function correctly is to open `demo.html` and ensure all icons load correctly.

Still another step is to create an s3 bucket on amazon that matches the arizona digital one and push to it using the aws cli using the commands in [.github/workflows/cdn-deploy-main.yml](https://github.com/az-digital/az-icons/pull/26/files#diff-0f4ec49e196161a346ad38516b6e57c58d0356e2f86f8c40d2a1a33606a7f88f)

## Types of changes
Devtools
CI

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+```
+npm run dist
+```
+
+See the [package.json](./package.json) for all of the available commands.
+
+## Pull Request Process
+1. Create an issue on GitHub with the appropriate [issue type template](https://github.com/az-digital/az-icons/issues/new/choose).
+2. Create a branch on the arizona-bootstrap repository using the issue type naming convention of `<issue-type>/<issue-number>`. Example: `feature/123` or `bug/123`.
```
Comment (trackleft) [inline CONTRIBUTING.md:93]:
```suggestion
2. Create a branch on the az-icons repository using the issue type naming convention of `<issue-type>/<issue-number>`. Example: `feature/123` or `bug/123`.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+See the [package.json](./package.json) for all of the available commands.
+
+## Pull Request Process
+1. Create an issue on GitHub with the appropriate [issue type template](https://github.com/az-digital/az-icons/issues/new/choose).
+2. Create a branch on the az-icons repository using the issue type naming convention of `<issue-type>/<issue-number>`. Example: `feature/123` or `bug/123`.
+3. Make and commit your changes.
+4. Create a pull request to be reviewed by the Arizona Digital community. Be sure to include a reference to the issue in your pull request description. Read more about [linking a pull request to an issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
+
+## Policies
+Contributions to Arizona Bootstrap must adhere to Arizona Digital's policies and [Code of Conduct](https://github.com/az-digital/policies/blob/main/code-of-conduct.md).
```
Comment (trackleft) [inline CONTRIBUTING.md:98]:
```suggestion
Contributions to Arizona Icons must adhere to Arizona Digital's policies and [Code of Conduct](https://github.com/az-digital/policies/blob/main/code-of-conduct.md).
```

File: package.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        "url": "https://github.com/az-digital/az-icons/issues"
+    },
+    "license": "UNLICENSED",
+    "homepage": "https://digital.arizona.edu/arizona-bootstrap/docs/2.0/icons",
+    "files": [
+        "dist/{svg,png,fonts}/*.{png,svg,woff,ttf,eot}",
+        "dist/*"
+    ],
+    "dependencies": {
+        "az-icons": "github:az-digital/az-icons#issues/3-contribution-workflow",
```
Comment (trackleft) [inline package.json:33]:
```suggestion
```

**Comment by akslay** [comment]
I encountered an error on the last step in your instructions: `npm install "https://github.com/az-digital/az-icons.git#issues/3-contribution-workflow" --save`

```
npm ERR! Error while executing:
npm ERR! /usr/bin/git ls-remote -h -t https://github.com/az-digital/az-icons.git.git
npm ERR! 
npm ERR! remote: Support for password authentication was removed on August 13, 2021. Please use a personal access token instead.
npm ERR! remote: Please see https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/ for more information.
npm ERR! fatal: Authentication failed for 'https://github.com/az-digital/az-icons.git.git/'
npm ERR! 
npm ERR! exited with error code: 128
```

**Review by akslay — DISMISSED** [review]

**Review by mmunro-ltrr — APPROVED** [review]
The local tests, using a current version of npm (v16.13.2) seemed to work, & the unminified css was identical to the current version at https://cdn.digital.arizona.edu/lib/az-icons/main/az-icons-styles.css

**Review by camikazegreen — APPROVED** [review]
It seems like the risk to merge this is minimal and acceptable.

**Review by smayden — APPROVED** [review]

**Pull request merged**